### ISSUE TITLE
[RFC]: add encoding and decoding functions in stdlib

### ISSUE THREAD

**Issue by adityacodes30** [stdlib-js/stdlib#2200]
### Description

This RFC proposes to add encoding and decoding functions to stdlib and possibly support multimodal inputs such as text , jpgs and more. This will have wide ranging implications when transferring data - example when working with cv we transfer video frames encoded.  I think this would be beneficial for a lot of other applications as well 

### Other

Seeking discussion on weather this should be done , if yes else would be looking to support 

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by kgryte** [comment]
@adityacodes30 Do you have specific examples regarding what you have in mind?

**Comment by adityacodes30** [comment]
yes , the first one that comes to mind is `base64` which can be particularly useful for encoding and image data , others that I think we should support are `base32 `  `huffman encoding`  some binary protocols such as` protobuf` would be good additions as well . Radix-64 will be a good plus as well

**Comment by kgryte** [comment]
@adityacodes30 Not opposed to pursuing things, but it would be good to see some concrete API proposals.

`base64` encode/decode utilities could be useful; `protobuf` we are less likely to support. The reason for the latter is that protocol is likely to continue evolving, which thus creates a maintenance burden for us to keep up.

Open to exploring image encoders/decoders. But again, would need to see concrete proposals, as some of these are likely to be quite involved, and it is not uncommon for these functionality to be written in C, so we'd need a browser implementation strategy (e.g., JS/WASM).

I suggest following up in this issue with a few concrete suggestions, as well as links to any reference implementations/prior art.
